### ISSUE TITLE
Depth uncertainty mask is computed but never applied

### ISSUE THREAD

**Issue by farukungit** [Liouvi/IG-SLAM#2]
## Summary

`GaussianModel.create_pcd_from_image` builds an uncertainty mask and then
applies it with `np.where(~mask, ...)`, but `get_depth_mask` returns an **int**
array. `~` on integers is a bitwise complement, so `~0 == -1` and `~1 == -2` —
both non-zero, hence both truthy for `np.where`. The condition is True at every
pixel and the depth map is returned unchanged.

The result is that the depth-uncertainty masking described in the paper
(Sec. 5.2) is inactive in the released code. Depth uncertainty still reaches the
loss through `depth_weights = sqrt(cov^-1)` in `src/mapping.py`, so the
contribution is not entirely lost, but no pixel is ever excluded from Gaussian
initialisation.

`gaussian_splatting/scene/gaussian_model.py:115-117`

```python
mask = get_depth_mask(cov, self.max_depth_sigma_thresh)   # -> int array
depthmap = np.where(~mask, depthmap, self.max_depth + 1.0)
```

`utils/mask_utils.py:45`

```python
return mask.astype(int)
```

## Reproduction

CPU only, numpy + scipy. Save as `repro.py` in the repository root and run
`python repro.py`.

```python
import numpy as np
from utils.mask_utils import get_depth_mask

H, W = 320, 640
yy, xx = np.mgrid[0:H, 0:W].astype(np.float32)
cov = 0.02 + 0.6 * np.exp(-(((xx - 200) ** 2 + (yy - 100) ** 2) / (2 * 80 ** 2)))
depth = np.full((H, W), 3.0)
MAX_DEPTH = 100.0

mask = get_depth_mask(cov, 0.2)
print("mask dtype              :", mask.dtype)
print("flagged uncertain       : %.1f%%" % (100 * (mask != 0).mean()))
print("~mask unique values     :", np.unique(~mask))

shipped  = np.where(~mask, depth, MAX_DEPTH + 1.0)
intended = np.where(~mask.astype(bool), depth, MAX_DEPTH + 1.0)
print("pixels actually dropped : %.1f%%" % (100 * (shipped  > MAX_DEPTH).mean()))
print("pixels that should drop : %.1f%%" % (100 * (intended > MAX_DEPTH).mean()))
```

Output on `main` (a508136):

```
mask dtype              : int32
flagged uncertain       : 36.1%
~mask unique values     : [-2 -1]
pixels actually dropped : 0.0%
pixels that should drop : 36.1%
```

## Second issue: zero-range normalisation

`get_depth_mask` normalises with `(cov - cov.min()) / (cov.max() - cov.min())`.
When the covariance map is uniform this divides by zero:

```
RuntimeWarning: invalid value encountered in divide
```

The resulting NaNs compare False against the threshold, and the two majority
filters then turn that into a spurious border ring (2252 flagged pixels on a
320x640 uniform map). This path is reachable in normal operation:
`DepthVideo` initialises `depths_cov` with `torch.ones` (`src/depth_video.py:56`)
and resets it to `ones_like` whenever the Cholesky factorisation fails
(`src/depth_video.py:306`).

## Note on fixing it

Simply returning a boolean array makes the mask live, which changes the
reconstruction produced by every released config. I have not been able to
validate that end to end — no GPU with enough memory here — so a PR is attached
that keeps the masking opt-in and leaves the default behaviour identical.

**Comment by Liouvi** [comment]
This code was written in 1 BC (Before Claude). I remember doing depth uncertainty mask ablations, and I might have broken the code while doing them. Thank you for noticing, I will fix it as soon as possible.
